### DISCUSSION TITLE
Use Workspace Create API

### DISCUSSION THREAD

**Discussion by whenSunSet** [coder/coder#7714] [General] [answered]
I want to pass some parameters to Terraform when using the Workspace create API, so that I can do some tasks when starting the container. How should I do this?
I am currently trying to use parameter_values and rich_parameter_values, but I don't know how to use them in the Terraform script.

**Comment by sharkymark** [comment, marked as answer]
@whenSunSet this worked for me.

[This is the template ](https://github.com/sharkymark/pandas_automl)that has a git repo url as the parameter

```sh
curl --request POST \
--url "$CODER_URL/api/v2/organizations/$ORG_ID/members/me/workspaces" \
--header "Coder-Session-Token: $API_KEY" \
  --data '{
  "name": "jupyter-lab",
  "rich_parameter_values": [
    {
      "name": "Git Repo URL",
      "value": "https://github.com/sharkymark/pandas_automl"
    }
  ],
  "template_id": "'$TEMPLATE_ID'"
  }'
```
